test.py

```diff
@@ -540,7 +468,6 @@ async def main() -> int:
     options = parse_cmd_line()
 
     open_log(options.tmpdir, f"test.py.{'-'.join(options.modes)}.log", options.log_level)
-    setup_cgroup(options.gather_metrics)
```

Comment:
You moved setup_cgroup() into prepare_dirs() which is not far away from here. This change looks isolated enough to come with its own patch. Can it?